site/docs/releases.md

```diff
@@ -88,6 +88,20 @@ The 1.8.1 release contains bug fixes and fixes to LICENSE/NOTICE files. For full
 
 ## Past releases
 
+### 1.7.2 release
+
+Apache Iceberg 1.7.2 was released on March 19, 2025.
+
+The 1.7.2 release contains bug fixes and new features. For full release notes visit [Github](https://github.com/apache/iceberg/releases/tag/apache-iceberg-1.7.2)
+
+* Core
+    - REST catalog doesn't return old history if we execute `CREATE OR REPLACE TABLE` statement ([\#11777](https://github.com/apache/iceberg/issues/11777))
```

Comment:
I would also add `Retain current view version during expiration` / `Fix possible deadlock in ParallelIterable` under `Core` and `Don't fetch credential from endpoint if properties contain a valid credential` under `AWS` as those are fixes worth mentioning

Reply:
Good point, let me add these fixes.